sys/kern/src/arch/arm_m.rs

```diff
@@ -1474,16 +1473,15 @@ unsafe extern "C" fn handle_fault(task: *mut task::Task) {
     // switch to a task to run.
     with_task_table(|tasks| {
         let next = match task::force_fault(tasks, idx, fault) {
-            task::NextTask::Specific(i) => i,
+            task::NextTask::Specific(i) => &tasks[i],
             task::NextTask::Other => task::select(idx, tasks),
-            task::NextTask::Same => idx,
+            task::NextTask::Same => &tasks[idx],
```

Comment:
It occurs to me that, because the length of the task table is always `HUBRIS_TASK_COUNT` and task indices are always in bounds, we could probably shave off a few instructions (and an unreachable panic site) by using `get_unchecked` when indexing the task table here and elsewhere in the scheduler. I'm not sure whether this optimization is worth the unsafe code, but it's worth thinking about...

Reply:
> we could probably shave off a few instructions (and an unreachable panic site) by using get_unchecked

That's probably true but we'd want to be _very_ careful about where we do it.

> why don't all these methods work with &(mut) [Task; HUBRIS_TASK_COUNT] references?

There's a historical reason why we don't use fixed-length arrays for the task table, but it's not really valid anymore. (The number of tasks was not originally available to the kernel at compile time.)

In practice, however, all of these functions get specialized by the compiler to fixed-length arrays. When disassembling the kernel you see a lot of `cmp r0, #6` and stuff for validation. Making the types explicit might be an interesting simplification -- relying on the compiler for stuff like this can be tricky.

I'd prefer not to pile either of those changes on _this_ PR if possible.

Reply:
Entirely reasonable on both counts, we can think more about this later.

Reply:
Agreed: Since I already have the changes locally, I'll try my hand at figuring out the asm output at play and see if this helps LLVM or just duplicates its work.

Reply:
Alright, I couldn't find any difference in output (except my mind or different checked out commit from this PR playing tricks on me).

I can still open a PR, if you'd like: The full patch file is only 434 lines long, with 44 actual lines changed I think. Very small, guarantees the optimisation and it might have _some_ effect somewhere, even though I was unable to find that place (I managed to only really analyse `select` and `safe_copy`.)